### PR TITLE
cls2-431-edit-task

### PR DESCRIPTION
**Pull request by chopkinsmade**
## Description of change

- Add a new edit task page component 

## Test instructions

Using any existing task, go to that page and click the edit button. Or use the url http://localhost:3000/investments/projects/abf2c1f4-0c43-4739-86e2-f9d12556a497/tasks/4a4376ed-e4a4-4298-96a7-9cd82c64375c/edit
Make a change and save the form. Go back to the edit page, and the new values will be showing

### After

screencapture-localhost-3000-tasks-f23a2689-0d0c-4dc2-bd0c-a45e4cec8969-2023-10-12-09_31_23

screencapture-localhost-3000-investments-projects-abf2c1f4-0c43-4739-86e2-f9d12556a497-tasks-f23a2689-0d0c-4dc2-bd0c-a45e4cec8969-edit-2023-10-12-09_29_53

## Checklist

[//]: # 'When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/main/docs/Code%20review%20guidelines.md'

- [ ] Has the branch been rebased to main?
- [ ] Automated tests (Any of the following when applicable: Unit, Functional or End-to-End)
- [ ] Manual compatibility testing (Browsers: Chrome, Firefox, Edge, Safari)

### CODE REVIEW

File: test/functional/cypress/specs/investments/project-edit-task-spec.js [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+      cy.get('[data-test="field-taskRemindersEnabled"]').then((element) => {
+        assertFieldRadiosWithLegend({
+          element,
+          legend: 'Do you want to set a reminder for this task?',
+          optionsCount: 3,
+          value: investmentProjectTask.emailRemindersEnabled
+            ? capitalize(OPTION_YES)
+            : capitalize(OPTION_NO),
+        })
+      })
```
Comment (marijnkampf) [inline test/functional/cypress/specs/investments/project-edit-task-spec.js:112]: Could this just be a check for the form and should the checks for the fields could be a component test instead?

**Comment by marijnkampf** [comment]
Editing a task created by another user re-assigns that tasks to the editor. Is this desired behaviour?

File: test/functional/cypress/specs/investments/project-edit-task-spec.js (hunk 1 above)
Comment (chopkinsmade) [reply test/functional/cypress/specs/investments/project-edit-task-spec.js:112]: Yeh that makes sense, I'll make that change

Comment (chopkinsmade) [reply test/functional/cypress/specs/investments/project-edit-task-spec.js:112]: Moved those into component, there is a test in the functional that changes some values in the form and asserts the api POST

**Comment by chopkinsmade** [comment]
> Editing a task created by another user re-assigns that tasks to the editor. Is this desired behaviour?

It's a bug I've fixed

File: src/client/modules/Investments/Projects/Tasks/transformers.js [hunk 2]
```diff
@@ -1,11 +1,7 @@
...
-import {
-  addDays,
-  addMonths,
-  formatWithoutParsing,
-  transformValueForAPI,
-} from '../../../../utils/date'
+import { OPTION_NO, OPTION_YES } from '../../../../../apps/constants'
+import { idNamesToValueLabels } from '../../../../utils'
+import { convertDateToFieldDateObject } from '../../../../utils/date'
+import { transformTaskFormValuesForAPI } from '../../../Tasks/TaskForm/transformers'
```
Comment (paulgain) [inline src/client/modules/Investments/Projects/Tasks/transformers.js:4]: This is optional but I think imports are easier to read when the longest in length are at the top. Sure, sometimes we group them by type but on this instance there are no groupings.
```javascript
import { transformTaskFormValuesForAPI } from '../../../Tasks/TaskForm/transformers'
import { OPTION_NO, OPTION_YES } from '../../../../../apps/constants'
import { convertDateToFieldDateObject } from '../../../../utils/date'
import { idNamesToValueLabels } from '../../../../utils'
```

File: src/client/modules/Investments/Projects/Tasks/transformers.js [hunk 3]
```diff
@@ -16,27 +12,17 @@ export const transformFormValuesForAPI = (
...
+  id: task.id,
+  investmentProject: task.investmentProjectTask.investmentProject,
+  taskTitle: task.title,
+  taskDescription: task.description,
+  taskDueDate: task.dueDate ? 'custom' : 'none',
+  customDate: task.dueDate ? convertDateToFieldDateObject(task.dueDate) : null,
+  taskRemindersEnabled: task.emailRemindersEnabled ? OPTION_YES : OPTION_NO,
+  taskReminderDays: task.reminderDays,
+  taskAdvisers: idNamesToValueLabels(task.advisers),
+})
```
Comment (paulgain) [inline src/client/modules/Investments/Projects/Tasks/transformers.js:28]: Nice 👍

**Review by paulgain — APPROVED** [review]

File: src/client/modules/Investments/Projects/Tasks/transformers.js (hunk 2 above)
Comment (chopkinsmade) [reply src/client/modules/Investments/Projects/Tasks/transformers.js:4]: Will rebase this onto the next PR and apply there

**Pull request merged**